src/Cake.Hosts/HostsFileUtil.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Cake.Hosts
+{
+    /// <summary>
+    /// Validate and extract data from hosts file
+    /// </summary>
+    public class HostsFileUtil
+    {
+        /// <summary>
+        /// Verify if line is from hosts file
+        /// </summary>
+        /// <param name="line">The line from hosts file</param>
+        /// <returns>If is not empty and not starts with #, is valid</returns>
+        static public bool IsLineAHostFile(string line) 
```

Comment:
More appropriate name is `IsLineAHostsRecord`

Reply:
Also, can we use the regex to do the matching, rather than just check that it does not start with `#`?